### PR TITLE
docs: improves documentation for components from l to n CNS-3985

### PR DESCRIPTION
**Pull request by vincentaudebert**
## Type

- Documentation

#### What is expected?

Documentation for L and N components

#### The following changes where made:

    Label
    Link
    MarkDown
    MenuIcon
    Modal
    Notice
    NotificationBar

### CODE REVIEW

File: src/components/Modal/__stories__/helpers.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+export const animations = [
```
Comment (chambo-e) [inline src/components/Modal/__stories__/helpers.js:1]: I think that you can export all these const from the component, there is going to be a discrepancy as some point if we go this way :)

**Review by chambo-e — CHANGES_REQUESTED** [review]

**Review by matthprost — CHANGES_REQUESTED** [review]
Before any `<ArgsTable>` can you add  `## API` title ?

File: src/components/MarkDown/__stories__/index.stories.mdx [hunk 2]
```diff
@@ -10,26 +16,29 @@ import { simple, withMultipleParagraphs } from './markdown_example'
...
-  <Story name="with-paragraphs">
+<Canvas>
+  <Story name="With paragraphs">
     <MarkDown source={withMultipleParagraphs} />
   </Story>
-  <Story name="with-paragraphs-inline">
+</Canvas>
+
+<Canvas>
+  <Story name="With paragraphs inline">
```
Comment (matthprost) [inline src/components/MarkDown/__stories__/index.stories.mdx:39]: Can we change it for `Paragraphs` and remove the `with`

**Review by matthprost — CHANGES_REQUESTED** [review]

**Comment by matthprost** [comment]
Screenshot 2021-05-31 at 10 13 13

No example for link component ?

File: src/components/MarkDown/__stories__/index.stories.mdx [hunk 3]
```diff
@@ -10,26 +16,29 @@ import { simple, withMultipleParagraphs } from './markdown_example'
...
     <MarkDown source={withMultipleParagraphs} inline />
   </Story>
 </Canvas>
+
```
Comment (matthprost) [inline src/components/MarkDown/__stories__/index.stories.mdx:43]: Both properties `linkTarget` and `escapeHtml ` are missing in examples and no comments on them so we don't know what they can be used for

**Review by matthprost — CHANGES_REQUESTED** [review]

File: src/components/MenuIcon/__stories__/index.stories.mdx [hunk 4]
```diff
@@ -12,12 +12,12 @@ import Box from '../../Box'
...
+<Description>
+  MenuIcon is an icon component styled to fit into menus
+</Description>
 
-<Description>MenuIcon //:TODO</Description>
-
-<Canvas isColumn>
-  <Story name="basic">
+<Canvas>
+  <Story name="Basic">
```
Comment (matthprost) [inline src/components/MenuIcon/__stories__/index.stories.mdx:20]: Many properties missing in examples such as `color`, `size` and `isButton`

**Review by matthprost — CHANGES_REQUESTED** [review]

**Review by chambo-e — APPROVED** [review]

**Review by matthprost — APPROVED** [review]

**Pull request merged**